proto/kava/liquidstaking/v1beta1/tx.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+package kava.liquidstaking.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/kava-labs/kava/x/liquidstaking/types";
+
+// Msg defines the liquidstaking Msg service.
+service Msg {
+
+  // MintDerivative defines a method for minting deriviatives
+  rpc MintDerivative(MsgMintDerivative) returns (MsgMintDerivativeResponse);
+}
+
+// MsgMintDerivative defines the Msg/MintDerivative request type.
+message MsgMintDerivative {
+  string sender    = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  string validator = 2;
+  bytes  shares = 3 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
```

Comment:
I was thinking through whether we should specify shares vs ukava in msgs. I think specifying ukava is slightly better. eg I'd change `shares` to `repeated Coin amount`.

User's think about their staked balance in terms of ukava. So they'd specify units of ukava when converting to derivatives. These ukava units need to be converted to share units somewhere (either front end or backend). Backend is probably a better place for any subtle math.

When we mint derivatives I think rounding the share amounts to 6 digits is slightly better than multiplying by 10^18. 18 digit precision coins are a bit of a pain: have to be more careful math funcs don't overflow, and they're harder to display in UIs. Downside is up to 1ukava loss when someone burns their derivatives. But even at 1000x the price that's still on the order of gas fees.